### PR TITLE
logging: support for file based logging

### PR DESCRIPTION
**Pull request by opreab**
Added basic file logging support
Added unittest for basicConfig when using files
Added stress test for logging to file and std stream

### CODE REVIEW

File: logging/example_logging.py [hunk 1]
```diff
@@ -1,9 +1,40 @@
+import os
+import time
 import logging
 
+
+def reset_log(log_file):
+    try:
+        os.remove(log_file)
+    except OSError:
+        pass
```
Comment (tomasz2101) [inline logging/example_logging.py]: Is it good approach to silently pass error?

File: logging/example_logging.py [hunk 2]
```diff
@@ -1,9 +1,40 @@
...
+    file_delta = time.time() - start_time
+    reset_log(log_file)
+
+    logging.basicConfig(filename=None)
+    start_time = time.time()
+    for i in range(iterations):
+        logging.info("%d %s" % (i, sample))
+    stdout_delta = time.time() - start_time
+    print("File logging time %f for %i iterations" % (file_delta, iterations))
+    print("Stdout logging time %f for %i iterations" % (stdout_delta, iterations))
```
Comment (tomasz2101) [inline logging/example_logging.py]: Why do you want to put this function in example ?

File: logging/logging.py [hunk 3]
```diff
@@ -38,11 +38,16 @@ def isEnabledFor(self, level):
...
+                _log_file.write(("%s:%s:" + msg + "\n") % ((self._level_str(level), self.name) + args))
+                _log_file.flush()
             else:
-                print(msg % args, file=_stream)
+                _stream.write("%s:%s:" % (self._level_str(level), self.name))
+                if not args:
+                    print(msg, file=_stream)
+                else:
+                    print(msg % args, file=_stream)
+
```
Comment (tomasz2101) [inline logging/logging.py]: Can we write to stream and file in same time?

**Comment by jimmo** [comment]
File based logging was added in #507.

**Pull request closed without merging**